data/campaigns/World_Conquest/scenarios/WC_II_scenario.cfg

```diff
@@ -22,17 +22,64 @@ _ "World Conquest 3p" #enddef
     _ "A randomly generated cooperative campaign for 3 players. It has 6 level of difficulties.
 (Expert level, 5 scenarios.)" #enddef
 
+#define WC2_HUMAN_DIFFICULTY IMAGE COLOR
+"units/{IMAGE}.png~RC(magenta>{COLOR})"#enddef
+
+#define WC2_NIGHTMARE_DIFFICULTY
+"units/monsters/fire-dragon.png~CROP(0,0,160,160)~SCALE_INTO(72,72)~RC(magenta>red)"#enddef
+
+#define WC2_CAMPAIGN_DIFFICULTY DIFFICULTY IMAGE LABEL DESCRIPTION POWER TRAINED HEROES GOLD TRAIN XP
+{CAMPAIGN_DIFFICULTY {DIFFICULTY} {IMAGE} {LABEL} {DESCRIPTION}}
+[event]
+    name=preload
+    # These events run for every difficulty, so this Lua block ensures only the active difficulty's
+    # event actually fills the settings variable.
+    [lua]
+        code="if wesnoth.scenario.difficulty ~= '{DIFFICULTY}' then" + <<
+            wesnoth.wml_actions["return"]{}
+        end>>
+    [/lua]
+    [set_variables]
+        name=wc2_difficulty
+        [value]
+            name={LABEL}
+            enemy_power={POWER}
+            enemy_trained={TRAINED}
+            heroes={HEROES}
+            # adjust bonus gold for number of players
+            extra_gold=$(gold * 2 ^ (3 - nplayers) where nplayers = $wc2_player_count, gold = {GOLD})
+            extra_training={TRAIN}
+            experience_penalty={XP}
+        [/value]
+    [/set_variables]
+[/event]
+#enddef
+
 #define WC2_CAMPAIGN_NEW PLAYERS
+    [campaign]
+        id = "WC_II_{PLAYERS}p{ID_SUFFIX}"
+        define = CAMPAIGN_WC
+        name= {WC_II_CAMPAIGN_NAME_{PLAYERS}P} + "{ID_SUFFIX}"
+        description = {WC_II_CAMPAIGN_DESC_{PLAYERS}P}
+        first_scenario = "WC_II_{PLAYERS}p{ID_SUFFIX}"
+        type = mp
+        abbrev = _ "WC"
+        {WC2_CAMPAIGN_DIFFICULTY VERY_EASY {WC2_HUMAN_DIFFICULTY human-peasants/peasant purple} _"Peasant" _"Very easy" 6 2 2 10 yes 0}
+        {WC2_CAMPAIGN_DIFFICULTY EASY {WC2_HUMAN_DIFFICULTY human-loyalists/sergeant black} _"Sergeant" _"Easy" 7 3 2 7 yes 5}
+        {WC2_CAMPAIGN_DIFFICULTY NORMAL {WC2_HUMAN_DIFFICULTY human-loyalists/lieutenant brown} _"Lieutenant" _"Normal" 8 4 2 5 yes 10} {DEFAULT_DIFFICULTY}
+        {WC2_CAMPAIGN_DIFFICULTY HARD {WC2_HUMAN_DIFFICULTY human-loyalists/general orange} _"General" _"Hard" 8 5 2 2 no 13}
+        {WC2_CAMPAIGN_DIFFICULTY VERY_HARD {WC2_HUMAN_DIFFICULTY human-loyalists/marshal white} _"Grand Marshal" _"Very hard" 9 6 2 1 no 17}
+        {WC2_CAMPAIGN_DIFFICULTY NIGHTMARE {WC2_NIGHTMARE_DIFFICULTY} _"Nightmare" _"Expert" 9 7 1 0 no 20}
```

Comment:
The rest I assume are existing strings, but are "Very easy" and "Very hard" already existing?

Reply:
Oh right, there's these too.

The one I was thinking of was `abbrev=_"WC"` but yeah, I guess there's three new strings so far.

Even though the others _are_ existing strings as far as I know, some of them might not be in the WC textdomain specifically, so I'll need to make sure to add some #textdomain statements if this is to go into 1.16.0.

Reply:
can "very easy" and "very hard" be renamed to "Novice/Rookie/Beginner" and "Challenging/Difficult" respectively?

Reply:
Sure.